### PR TITLE
feat(motion_velocity_smoother): use resampling function from motion_utils

### PR DESCRIPTION
**Pull request by purewater0901**
## Description

Use motion utils functions in motion velocity smoother(behavior velocity also has to change) and obstacle cruise planner

image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by purewater0901** [comment]
Backward driving also works fine
https://user-images.githubusercontent.com/43805014/182283934-8fc5a359-6e1d-4920-b779-b3093faf4311.mp4

**Comment by taikitanaka3** [comment]
LGTM
wait until CI with this branch has same successful rate as before

File: planning/obstacle_cruise_planner/src/node.cpp [hunk 1]
```diff
@@ -75,57 +76,15 @@ bool isFrontObstacle(
...
-  return output;
-}
-
-// TODO(murooka) replace with spline interpolation
 Trajectory decimateTrajectory(const Trajectory & input, const double step_length)
 {
   Trajectory output{};
 
-  if (input.points.empty()) {
+  if (input.points.size() < 2) {
```
Comment (taikitanaka3) [inline planning/obstacle_cruise_planner/src/node.cpp:83]: @purewater0901 
is this change intentional？

File: planning/motion_velocity_smoother/src/smoother/analytical_jerk_constrained_smoother/analytical_jerk_constrained_smoother.cpp [hunk 2]
```diff
@@ -291,19 +295,17 @@ boost::optional<TrajectoryPoints> AnalyticalJerkConstrainedSmoother::applyLatera
   for (double s = 0; s < in_arclength.back(); s += points_interval) {
     out_arclength.push_back(s);
   }
-  auto output = trajectory_utils::applyLinearInterpolation(in_arclength, input, out_arclength);
-  if (!output) {
-    RCLCPP_WARN(logger_, "Interpolation failed at lateral acceleration filter.");
-    return boost::none;
```
Comment (taikitanaka3) [inline planning/motion_velocity_smoother/src/smoother/analytical_jerk_constrained_smoother/analytical_jerk_constrained_smoother.cpp:297]: @purewater0901 
I think case output none is necesary.

**Comment by purewater0901** [comment]
evaluator result

https://evaluation.tier4.jp/evaluation/reports/862cde58-f1ef-5eed-9b62-b0d53034e8a1?project_id=prd_jt

File: planning/obstacle_cruise_planner/src/node.cpp (hunk 1 above)
Comment (purewater0901) [reply planning/obstacle_cruise_planner/src/node.cpp:83]: Yes, this is for suppressing the warning from the resampling functions in motion utils

File: planning/motion_velocity_smoother/src/smoother/analytical_jerk_constrained_smoother/analytical_jerk_constrained_smoother.cpp (hunk 2 above)
Comment (purewater0901) [reply planning/motion_velocity_smoother/src/smoother/analytical_jerk_constrained_smoother/analytical_jerk_constrained_smoother.cpp:297]: New resampling functions from motion utils do not use boost::optional, therefore, we do not need to handle none output

**Review by taikitanaka3 — APPROVED** [review]
LGTM
